src/browser/provider/pool.js

```diff
@@ -0,0 +1,145 @@
+var npmList = null;
+
+
+const BROWSER_PROVIDER_RE = /^(?:([^:]+):)?(.*)$/;
+const PROVIDER_MODULE_RE  = /^testcafe-provider-/;
+
+const providersCache  = {
+    'local':  new BrowserProviderHost(LocalBrowserProvider),
+    'path':   new BrowserProviderHost(PathBrowserProvider),
+    'remote': new BrowserProviderHost(RemoteBrowserProvider),
+    'sauce':  new BrowserProviderHost(SaucelabsBrowserProvider)
+};
+
+
+export default {
+    initialized: false,
+
+    async _handlePathAndCmd (browserDesc) {
+        var pathProvider = this.getProvider('path');
+        var options      = browserDesc.cmd ? { cmd: browserDesc.cmd } : null;
+
+        return [{ provider: pathProvider, alias: browserDesc.path, options }];
+    },
+
+    async _parseAlias (browserDesc) {
+        var providerRegExpMatch = BROWSER_PROVIDER_RE.exec(browserDesc);
+
+        if (!providerRegExpMatch)
+            throw new GeneralError(MESSAGE.cantFindBrowser, browserDesc);
+
+        var providerName = providerRegExpMatch[1];
+        var alias        = providerRegExpMatch[2];
+
+        var provider = this.getProvider(providerName || 'local');
+
+        if (alias === 'all') {
+            var allAliases = await provider.listAvailableAliases();
+
+            return allAliases.map(al => ({ provider, alias: al }));
+        }
+
+        if (!await provider.isValidAlias(alias))
+            throw new GeneralError(MESSAGE.cantFindBrowser, browserDesc);
+
+        return [{ provider, alias }];
+    },
+
+    async _parseBrowserDescription (browserDesc) {
+        if (browserDesc && browserDesc.path)
+            return this._handlePathAndCmd(browserDesc);
+
+        if (typeof browserDesc === 'string')
+            return this._parseAlias(browserDesc);
+
+        throw new GeneralError(MESSAGE.cantFindBrowser, browserDesc);
+    },
+
+    async getBrowserInfo (browserDesc) {
+        if (browserDesc instanceof BrowserConnection)
+            return browserDesc;
+
+        var browserInfo = await this._parseBrowserDescription(browserDesc);
+
+        await Promise.all(browserInfo.map(browser => {
+            return browser.provider.getDescriptorDefaultValues(browser.alias)
+                .then(descriptor => {
+                    assignIn(descriptor, browser.options);
+
+                    browser.descriptor = descriptor;
+                });
+        }));
+
+        return browserInfo;
+    },
+
+    async updateProvidersCache (searchGlobal = false) {
+        var prevNpmGlobal = npm.config.get('global');
+
+        npm.config.set('global', searchGlobal);
+
+        var packageData = await npmList([], true);
+
+        var providersModules = Object
+            .keys(packageData.dependencies)
+            .filter(moduleName => PROVIDER_MODULE_RE.test(moduleName))
+            .map(moduleName => ({ name: moduleName.replace(PROVIDER_MODULE_RE, ''), provider: require(moduleName) }));
+
+        providersModules.forEach(({ name, provider }) => this.addProvider(name, provider));
```

Comment:
Does it search only for globally installed modules? Because we need to find locally installed modules as well (if it's possible)